### ISSUE TITLE
Add file filter on /docker-entrypoint-initdb.d scripts

### ISSUE THREAD

**Issue by joraff** [docker-library/mysql#265]
Initializing a container results in `ERROR: Can't initialize batch_readline - may be the input source is a directory or a block device.` when a mounted file is missing from the host due to docker's behavior to create an empty directory for any missing mounted file.

```
root@8238788eac51:/# ls -la /docker-entrypoint-initdb.d/
total 8
drwxr-xr-x  3 root root 4096 Feb 15 19:09 .
drwxr-xr-x 58 root root 4096 Feb 15 19:32 ..
drwxr-xr-x  2 root root   68 Feb 15 19:09 database.sql
root@8238788eac51:/# for f in /docker-entrypoint-initdb.d/*; do echo $f; done                                                           
/docker-entrypoint-initdb.d/database.sql
```

Can a simple file test be added here?
https://github.com/docker-library/mysql/blob/9898215028f0b70d70ab4cc28d5136bdb7bed8a4/5.5/docker-entrypoint.sh#L165

```bash
for f in $(find /docker-entrypoint-initdb.d/* -type f); do
    case "$f" in
            *.sh)     echo "$0: running $f"; . "$f" ;;
            *.sql)    echo "$0: running $f"; "${mysql[@]}" < "$f"; echo ;;
            *.sql.gz) echo "$0: running $f"; gunzip -c "$f" | "${mysql[@]}"; echo ;;
            *)        echo "$0: ignoring $f" ;;
        esac
    echo
done
```

or 

```bash
for f in /docker-entrypoint-initdb.d/*; do
    if [ ! -f $f ]; then
      echo "ignoring $f: not a regular file";
      continue
    fi
    case "$f" in
            *.sh)     echo "$0: running $f"; . "$f" ;;
            *.sql)    echo "$0: running $f"; "${mysql[@]}" < "$f"; echo ;;
            *.sql.gz) echo "$0: running $f"; gunzip -c "$f" | "${mysql[@]}"; echo ;;
            *)        echo "$0: ignoring $f" ;;
        esac
    echo
done
```
---
Edit: reorganized paragraphs

**Comment by joraff** [comment]
I am now mounting a whole folder to /docker-entrypoint-initdb.d rather than a file, which I like better, but there could be a use case where a single file is easier to link.

**Comment by yosifkit** [comment]
I like option two with `if [ ! -f $f ]`, though not certain on what to print out when skipping.

**Comment by qrkourier** [comment]
I use the initdb.d directory as a docker volume to allow scripts to be run on a pristine container image to seed data for integration tests. When dropping root privs to become user mysql the host's seed scripts mounted on that directory are not necessarily readable, so I've refactored the entrypoint script in [this pull request](https://github.com/docker-library/mysql/pull/429/files) to first copy those files to /etc/mysql/initdb.d, chown, and then drop privileges. I also added a bugfix (scraped from another repo's identical issue) for the intermittent failure of GnuPG to fetch the pubkey that occurs when the keyserver is not available, or too slow.
